config/src/integration-test/java/org/springframework/security/config/annotation/authentication/ldap/LdapAuthenticationProviderBuilderSecurityBuilderTests.java

```diff
@@ -0,0 +1,250 @@
+import org.springframework.context.annotation.Import;
+import org.springframework.ldap.core.support.BaseLdapPathContextSource;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.authentication.configuration.EnableGlobalAuthentication;
+import org.springframework.security.config.annotation.configuration.ObjectPostProcessorConfiguration;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.test.SpringTestRule;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.authority.mapping.GrantedAuthoritiesMapper;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.ldap.DefaultSpringSecurityContextSource;
+import org.springframework.security.ldap.authentication.LdapAuthenticationProvider;
+import org.springframework.security.ldap.server.ApacheDSContainer;
+import org.springframework.security.ldap.userdetails.LdapAuthoritiesPopulator;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.List;
+
+import static java.util.Collections.singleton;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestBuilders.formLogin;
+import static org.springframework.security.test.web.servlet.response.SecurityMockMvcResultMatchers.authenticated;
+
+public class LdapAuthenticationProviderBuilderSecurityBuilderTests {
+	@Rule
+	public final SpringTestRule spring = new SpringTestRule();
+
+	@Autowired
+	private MockMvc mockMvc;
+
+	@Autowired
+	private AuthenticationManager authenticationManager;
+
+	@Test
+	public void defaultConfiguration() {
+		this.spring.register(DefaultLdapConfig.class).autowire();
+		LdapAuthenticationProvider provider = ldapProvider();
+
+		LdapAuthoritiesPopulator authoritiesPopulator = getAuthoritiesPopulator(provider);
+		assertThat(authoritiesPopulator).hasFieldOrPropertyWithValue("groupRoleAttribute", "cn");
+		assertThat(authoritiesPopulator).hasFieldOrPropertyWithValue("groupSearchBase", "");
+		assertThat(authoritiesPopulator).hasFieldOrPropertyWithValue("groupSearchFilter", "(uniqueMember={0})");
+		assertThat(ReflectionTestUtils.getField(getAuthoritiesMapper(provider), "prefix")).isEqualTo("ROLE_");
+	}
+
+	@EnableWebSecurity
+	static class DefaultLdapConfig extends BaseLdapProviderConfig {
+		// @formatter:off
+		protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+			auth
```

Comment:
The expected indentation here is
```
auth
	.ldapAuthentication()
		.contextSource(contextSource())
		.userDnPatterns("uid={0},ou=people");
```
The convention is that each layer of configuration properties is indented further.
In this case, since `contextSource` and `userDnPatterns` are methods of `ldapAuthentication`, then they should be indented with an additional tab.
Similar for all the other configurations.

Reply:
Ok, i fixed the indentation of the configurations